lib/TPP/IR/MatcherUtils.cpp

```diff
@@ -112,10 +108,8 @@ std::pair<bool, bool> isBrgemmVnniOp(linalg::GenericOp linalgOp,
     return std::make_pair(false, hasBatch);
   }
 
-  auto vnniDim =
-      vnni::utils::isInVnniLayout(linalgOp, mapOperandB, *blockingFactor);
-  bool isBrgemmOp = succeeded(vnniDim) && vnniDim->getPosition() == kRedIter;
-  return std::make_pair(isBrgemmOp, hasBatch);
+  bool isBrgemmVnni = vnni::utils::isInVnniLayout(linalgOp, *blockingFactor);
```

Comment:
If this is guaranteed to be BRGEMM (previous checks), then at least a comment saying so would help

Reply:
This function got confusing as it also checks for GEMM. Refactored a bit to improve readability.